manifests/timersengine.pp

```diff
@@ -0,0 +1,81 @@
+# @summary Manages the <code>st2timersengine</code> service.
+#
+# Normally this class is instantiated by +st2::profile::fullinstall+.
+# However, advanced users can instantiate this class directly to configure
+# and manage just the <code>st2timersengine</code> service on a single node.
+# Parameters for this class mirror the parameters in the st2 config.
+#
+# @see https://github.com/StackStorm/st2/blob/master/conf/st2.conf.sample#L337-L343
+#
+# @example Basic usage
+#   include ::st2::timersengine
+#
+# @example Customizing parameters
+#   class { '::st2::timersengine':
+#     enabled  => true,
+#     timezone => 'America/Los_Angeles',
+#   }
+#
+# @param enabled
+#   Specify to enable timer service.
+# @param timezone
+#   Timezone pertaining to the location where st2 is run.
+#
+class st2::timersengine (
+  $enabled  = $::st2::timersengine_enabled,
+  $timezone = $::st2::timersengine_timezone,
+) inherits st2 {
+
+  # st2timersengine was introduced in 2.9.0
+  if ($::st2::version == 'latest' or
+      $::st2::version == 'present' or
+      $::st2::version == 'installed' or
```

Comment:
In other words, if st2 version is installed (and not latest), how do we know if installed version older or newer then `2.9.0` and need to enable/disable services like timersengine?

Reply:
`$::st2::version`  is specified here: https://github.com/StackStorm/puppet-st2/blob/master/manifests/init.pp#L121

By default it is `present` and we assume that is the newest installed version. If the user wishes to pin to a specific version they can specify the parameter to be something like `'2.9.1'`.

Reply:
What puppet means by `ensure => present` for its packages is that it will install the newest version when installing fresh. Otherwise if you specify a specific version `ensure => '2.9.1'` it will install only that specific version.

Reply:
Right, as I understand `present` and `installed` means that the requirement is to have "any" version installed in the system, not necessarily latest.
Meaning with `present` or `installed` the package won't get an update if it's already exists in the system.

With that logic, if older version of the package is already installed we can mistakenly assume it's "latest" and enable timersengine which will fail because of older st2 version.